Vizzuality/heco-invest
# Add UserMenu and NavigationMenuButton to the Discover/Search layout

## NOTE

This PR is built on top of #159. 

## Description

This PR Adds the UserMenu and Menu button to the header of the Discover layout. 

In addition, I've split the menus into their own separate components, so that they can be reused in different layouts' headers. 

&nbsp; 

**Note:**  
I ran into `z-index` issues. Due to how the layout was built, the `DiscoverSearch` overlapping the header bar needed to have a higher `z-index` than the header bar. However, due to how the menu `Popover` is built, this would mean they would be displayed _below_ the `DiscoverSearch` component. 

I think a proper solution would to be rebuilding the menu popovers to be broken free from its parent component; I gave it a  quick try with [portals](https://reactjs.org/docs/portals.html) & [popper](https://popper.js.org/react-popper/v2/), however I didn't manage to pull it off in a timely manner. 

As a quick (but perhaps not ideal) workaround, `DiscoverSearch` was also added to the `Header`. Maybe we can revisit this in the future. 

## Testing instructions

Verify that:  
- The user menu displays correctly (and above the `DiscoverSearch` component)  
- The menu button displays correctly  
- The user menu is only visible for `md` and larger screens (the menu button turns into a hamburguer button on small screens and takes over)  
- When the user is not signed in, a Sign In button is shown instead of the user menu  

## Tracking

[LET-439](https://vizzuality.atlassian.net/browse/LET-439)


## Screenshots 
<img width="497" alt="Screenshot 2022-05-16 at 15 48 23" src="https://user-images.githubusercontent.com/6273795/168621255-6735d23a-3f83-4a9b-a056-e13a75245c35.png">

<img width="633" alt="Screenshot 2022-05-16 at 15 48 33" src="https://user-images.githubusercontent.com/6273795/168621197-ff65ceea-893e-45f1-a346-46e59a49f682.png">
